openapi_core/wrappers.py

```diff
@@ -107,6 +107,45 @@ def mimetype(self):
         return self.request.mimetype
 
 
+class RequestsOpenAPIRequest(BaseOpenAPIRequest):
+    def __init__(self, request):
+        self.request = request
+        self.url = urlparse(request.url)
+
+    @property
+    def host_url(self):
+        return self.url.scheme + '://' + self.url.netloc
+
+    @property
+    def path(self):
+        return self.url.path
+
+    @property
+    def method(self):
+        return self.request.method.lower()
+
+    @property
+    def path_pattern(self):
+        return self.url.path
```

Comment:
I don't think your implementation for the path_pattern is going to work for routes that have path parameters. The library does a dictionary lookup where it expects the path_pattern to equal the path in the spec. For example, consider the path '/pets/{petId}'. Your implementation will cause the library to try to look up something like '/pets/123' in the paths dict, which will cause the validation to fail because that's not the way the path is defined in the OAS spec.

Reply:
I'll be pushing a PR sometime today that should deal with the path_pattern problem. If accepted then your implementation will likely work correctly with my addition.

Reply:
Actually, it's more complicated than I thought. I have something that would work for the path_pattern but it doesn't account for the path and query parameters.

Reply:
Yes, if we have route pattern, this would not work. I think we should add route pattern when creating RequestsOpenAPIRequest object:

    openapi_request = RequestsOpenAPIRequest(req, url_pattern='/pets/{petId}')

Currently I am working on it, could not say when it would ready...

Reply:
I'm wondering if this should actually be a part of the library. Consider this.

We already have a regex-like pattern in the route:
```/pets/{petId}) ```
If we convert this into a true regex then we should be able to match against every possible pattern that this path can have:
/pets/(.*)
It makes sense to do this in the spec object rather than the request object because the spec contains the patterns of every possible path in the API. Thus we merely match the url against all possible paths, which is O(n) though n should be small, and pick the pattern that matches the raw url.

The problem then (and this is where I am at the moment) is that we must also extract the path path and query parameters. The query params should be able to be parsed easily. The path params require a bit more work since this is part of the validators, but it's definitely doable.

A second option is to pass the spec to the request object and let it figure out everything there. However, I feel it would be far more user friendly to simply be able to have the core library extract this information from a simpler request object since there are plenty or url parsing tools and it would make implementing request wrappers much easier in the future.